plugin/README.md

```diff
@@ -0,0 +1 @@
+After making any changes to `index.ts` make sure to run `tsc` afterwards to get expected `index.js`!
```

Comment:
First, I would like this `README.md` to be analogous to the official Babel plugins from [here](https://github.com/babel/babel/tree/main/packages/babel-plugin-proposal-export-namespace-from).

Secondly, please provide a high-level overview of what the plugin does (preparing worklets including closure calculation, auto-workletization of Reanimated hooks and Gesture Handler callbacks).

Thirdly, this is a great place to gather some information for future users and developers of the plugin. In particular, let's document how to:
* configure the development environment
* run the plugin in dev environment (`cd react-native-reanimated && npx babel file.js`)
* debug the plugin using VSCode JavaScript debugger
* run tests (`cd react-native-reanimated && yarn jest plugin.jest.js` or even better: `cd plugin && yarn test`)
* check types (`cd plugin && yarn tsc --noEmit`)
* run linter (`cd plugin && yarn lint`)
* run formatter (`cd plugin && yarn format`)

Finally, please attach these scripts to the CI.

Reply:
While it is a great idea, I think we should wait for new documentation for Reanimated3 with it. We could provide instructions on the website and simply link it in this readme.

Reply:
I agree with @tomekzaw that the readme is the correct place for this information.
The docs are meant for Reanimated users, and should instruct how to install the plugin, maybe provide some rationale why it's necessary, and that's basically it, while the information on plugin development belongs in its repo.

At least that's my 2 cents 😉